xrpl4j-core/src/main/java/org/xrpl/xrpl4j/crypto/signing/bc/BcSignatureService.java

```diff
@@ -127,8 +127,7 @@ protected synchronized Signature ecDsaSign(final PrivateKey privateKey, final Un
     Objects.requireNonNull(transactionBytes);
 
     final UnsignedByteArray messageHash = HashingUtils.sha512Half(transactionBytes);
-
-    final BigInteger privateKeyInt = new BigInteger(privateKey.value().toByteArray());
+    final BigInteger privateKeyInt = new BigInteger(privateKey.valueWithPrefixedBytes().toByteArray());
```

Comment:
TODO: Check

Reply:
I think this should probably be `privateKey.valueWithNaturalBytes`? Probably doesn't matter because the prefix will be 0x00 which won't affect the value of the BigInteger, but un-prefixed seems more correct